presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java

```diff
@@ -1100,6 +1102,16 @@ public SystemSessionProperties(
                         AGGREGATION_IF_TO_FILTER_REWRITE_ENABLED,
                         "Enable rewriting the IF expression inside an aggregation function to a filter clause outside the aggregation",
                         featuresConfig.isAggregationIfToFilterRewriteEnabled(),
+                        false),
+                booleanProperty(
+                        HEAP_DUMP_ON_EXCEEDED_MEMORY_LIMIT_ENABLED,
+                        "Trigger heap dump to `EXCEEDED_MEMORY_LIMIT_HEAP_DUMP_FILE_PATH` on exceeded memory limit exceptions",
+                        false,
+                        false),
+                stringProperty(
+                        EXCEEDED_MEMORY_LIMIT_HEAP_DUMP_FILE_PATH,
+                        "File to which heap snapshot will be dumped, if heap_dump_on_exceeded_memory_limit_enabled",
+                        System.getProperty("java.io.tmpdir") + "heap_dump.hprof",
```

Comment:
probably we wanna specify the directory path instead of the file name. Then we can encode our own file name like <query_id>_<stage_id>.hprof